src/views/forms/FormSubAccountCreation/FormSubAccountCreationTs.ts

```diff
@@ -233,25 +248,82 @@ export class FormSubAccountCreationTs extends Vue {
       )
       return null
     }
+    if (this.isLedger) {
+      this.importSubAccountFromLedger(childAccountName)
+        .then((res) => {
+          this.accountService.saveAccount(res)
+          // - update app state
+          this.$store.dispatch('profile/ADD_ACCOUNT', res)
+          this.$store.dispatch('account/SET_CURRENT_ACCOUNT', res)
+          this.$store.dispatch('account/SET_KNOWN_ACCOUNTS', this.currentProfile.accounts)
+          this.$store.dispatch('notification/ADD_SUCCESS', NotificationType.OPERATION_SUCCESS)
+          this.$emit('submit', this.formItems)
+        })
+        .catch((err) => console.log(err))
+    } else {
+      // - get next path
+      const nextPath = this.paths.getNextAccountPath(this.knownPaths)
+
+      this.$store.dispatch('diagnostic/ADD_DEBUG', 'Adding child account with derivation path: ' + nextPath)
+
+      // - decrypt mnemonic
+      const encSeed = this.currentProfile.seed
+      const passphrase = Crypto.decrypt(encSeed, this.currentPassword.value)
+      const mnemonic = new MnemonicPassPhrase(passphrase)
+
+      // create account by mnemonic
+      return this.accountService.getChildAccountByPath(
+        this.currentProfile,
+        this.currentPassword,
+        mnemonic,
+        nextPath,
+        this.networkType,
+        childAccountName,
+      )
+    }
+  }
 
-    // - get next path
-    const nextPath = this.paths.getNextAccountPath(this.knownPaths)
-
-    this.$store.dispatch('diagnostic/ADD_DEBUG', 'Adding child account with derivation path: ' + nextPath)
-
-    // - decrypt mnemonic
-    const encSeed = this.currentProfile.seed
-    const passphrase = Crypto.decrypt(encSeed, this.currentPassword.value)
-    const mnemonic = new MnemonicPassPhrase(passphrase)
-
-    // create account by mnemonic
-    return this.accountService.getChildAccountByPath(
-      this.currentProfile,
-      this.currentPassword,
-      mnemonic,
-      nextPath,
-      this.networkType,
-      childAccountName,
-    )
+  async importSubAccountFromLedger(childAccountName: string): Promise<AccountModel> | null {
+    const subAccountName = childAccountName
+    const accountPath = this.currentAccount.path
+    const currentAccountIndex = accountPath.substring(accountPath.length - 2, accountPath.length - 1)
+    const numAccount = this.knownPaths.length
+    let accountIndex
+    if (numAccount <= Number(currentAccountIndex)) {
+      accountIndex = numAccount + Number(currentAccountIndex)
+    } else {
+      accountIndex = numAccount + 1
+    }
```

Comment:
please use `DerivationService.incrementPathLevel`

Reply:
I see that currently, your account have path structure: 'm/44\'/4343\'/0\'/0\'/0\', then with derivationService.incrementPathLevel the next path will be 'm/44\'/4343\'/1\'/0\'/0\', but with hd-wallet, Ledger path structure is:  'm/44\'/4343\'/{networktype}\'/0\'/{index}\'', then next path will be increment the index. If I use derivationService here, the next path will increment in {nextworktype} position, it is wrong path. This is the reason I have to use the accountIndex here.